### DISCUSSION TITLE
🗳️ Open extension to other languages than C++ :snake: :coffee:

### DISCUSSION THREAD

**Discussion by adriens** [duckdb/duckdb#6899] [Polls]
Would it be interesting to be able to code extensions in other languages than C++ ?

**Comment by adriens** [comment]
it would make this one a piece of cake with python: 

- https://github.com/duckdb/duckdb/discussions/6872

**Reply by Mause** [reply]
Python is unlikely to be a suitable language to write an extension in, as you'd either need to write a wrapper in another language to embed the Python interpreter, or you'd need to compile the Python code to a native binary (specifically to a .so library or similar)

Neither of these options require us to implement special support for Python

**Reply by adriens** [reply]
And about java ?

**Reply by adriens** [reply]
...or GO ?

**Reply by Mause** [reply]
> And about java ?

Java has the same issue as Python, though perhaps worse, as I'm not sure if it's possible to embed a java runtime inside your own library

**Reply by Mause** [reply]
> ...or GO ?

Go is compilable to native libraries, so you can write an extension using it yes. I'm not sure if the community supported DuckDB go binding has added this functionality however

**Comment by carlopi** [comment]
I believe currently there are example of Rust extensions: https://github.com/Mause/duckdb-extension-framework or https://github.com/ywelsch/duckdb-prql

Currently some degree of interoperability with a C/C++ API is required, but especially if you target the C API as @Mause did it from a C-derived languages (say... Zig).
